src/encoded/submission_status.py

```diff
@@ -0,0 +1,148 @@
+from pyramid.view import view_config
+from snovault.util import debug_log
+from dcicutils.misc_utils import ignored
+from snovault.search.search import search
+from snovault.search.search_utils import make_search_subreq
+from urllib.parse import urlencode
+import pprint
+
+# Portal constants
+FILESET = "FileSet"
+UPLOADED = "uploaded"
+UPLOADING = "uploading"
+STATUS = "status"
+O2_PATH = "o2_path"
+SUBMITTED_FILE = "SubmittedFile"
+FILE_FORMAT = "file_format"
+DISPLAY_TITLE = "display_title"
+
+WASHU_GCC = "WASHU GCC"
+BCM_GCC = "BCM GCC"
+NYGC_GCC = "NYGC GCC"
+BROAD_GCC = "BROAD GCC"
+UWSC_GCC = "UWSC GCC"
+
+
+def includeme(config):
+    config.add_route("get_submission_status", "/get_submission_status/")
+    config.scan(__name__)
+
+
+@view_config(route_name="get_submission_status", request_method="POST")
+@debug_log
+def get_submission_status(context, request):
+
+    try: 
+        post_params = request.json_body
+        filter = post_params.get("filter")
+        # Generate search total
+        search_params = {}
+        search_params["type"] = FILESET
+        add_search_filters(search_params, filter)
+        num_total_filesets = search_total(context, request, search_params)
+
+        # Generate search
+        search_params = {}
+        search_params["type"] = FILESET
+        search_params["limit"] = min(post_params["limit"], 50)
+        search_params["from"] = post_params["from"]
+        search_params["sort"] = f"-date_created"
+        add_search_filters(search_params, filter)
+        subreq = make_search_subreq(
+            request, f"/search?{urlencode(search_params, True)}", inherit_user=True
+        )
+        search_res = search(context, subreq)["@graph"]
+        file_sets = []
+        for res in search_res:
+            file_set = res
+            file_set["submitted_files"] = process_files_metadata(res.get("files", []))
+            file_sets.append(file_set)
```

Comment:
Is the reformatting really necessary? This is the only thing I see here that worries me, you have to get all requested results and post pass them here. It would probably be more efficient to use a search generator so you can stream results.

Reply:
I am only getting 30 Filesets at a time (per POST request) and the items aren't very big. Loading of new filesets in the demo on wolf seems very quick. I agree that it could be more efficient, but it's probably not worth the effort? However, if you foresee issues with the current solution, I can optimize it of course.

Reply:
Yeah let's punt on this for now